scripts/setup-opam.sh

```diff
@@ -10,46 +16,23 @@ export OPAMYES=1
 # Set term to xterm if not set
 export TERM=${TERM:-xterm}
 
-SWITCH='ocaml-variants.4.07.1+logoom'
-
-if [[ -d ~/.opam ]]; then
-  # ocaml environment
+# init opam
+if ! [[ -d ~/.opam ]]; then
+  opam init
   eval $(opam config env)
-
-  # check for cache'd opam
-  SWITCH_LIST=$(opam switch list -s)
-
-  # Check to see if we have explicit switch version
-  SWITCH_FOUND=false
-  for val in $SWITCH_LIST; do
-    if [ $val == $SWITCH ]; then
-      SWITCH_FOUND=true
-    fi
-  done
-else
-  # if there is no opam switch initialized, start from scratch
-  SWITCH_FOUND=false
 fi
 
-pushd /home/opam/opam-repository && git pull && popd
+# create local switch
+opam switch import src/opam.export --switch .
+sudo chmod -R u+rw _opam
+eval $(opam config env)
 
-if [ "$SWITCH_FOUND" = true ]; then
-  # Add the o1-labs opam repository
-  opam repository add --yes --all --set-default o1-labs https://github.com/o1-labs/opam-repository.git
-  opam switch set $SWITCH
-else
-  # Build opam from scratch
-  opam init
-  # Add the o1-labs opam repository
-  opam repository add --yes --all --set-default o1-labs https://github.com/o1-labs/opam-repository.git
```

Comment:
we can add this repository every time, it'll just do nothing if it's already added.
Also I make sure we only add this repository to the current switch, not the default one.